### PR TITLE
feat(Rich Text Editor): slate to MD transformer [#2xjpz9h]

### PR DESCRIPTION
**Pull request by bojangles-m**

### CODE REVIEW

File: src/components/RichTextEditor/serializer/markdown/__tests__/SlateToMarkdown.spec.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { SlateToMarkdown } from '..';
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/serializer/markdown/__tests__/SlateToMarkdown.spec.ts:3]:
```suggestion
import { SlateToMarkdown } from '../SlateToMarkdown';
```

File: src/components/RichTextEditor/serializer/markdown/types.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ export type BlockType = {
     children: NodeType[];
     parentType?: string;
     caption?: string;
-    language?: string;
+    language?: string | null;
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/serializer/markdown/types.ts:84]: We need to decide between `undefined` or `null` :p

File: src/components/RichTextEditor/serializer/markdown/__tests__/SlateToMarkdown.spec.ts (hunk 1 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/markdown/__tests__/SlateToMarkdown.spec.ts]: Updated

File: src/components/RichTextEditor/serializer/markdown/types.ts (hunk 2 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/markdown/types.ts]: Updated

Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/markdown/types.ts]: The null i get from passer could only transform to undefined :)

File: src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts [hunk 3]
```diff
@@ -131,6 +122,9 @@ export default function deserialize<T extends InputNodeTypes>(node: MarkdownAstN
     }
 }
 
+const hasNodeChildren = (node: MarkdownAstNode): boolean =>
+    !!node.children && Array.isArray(node.children) && node.children.length > 0;
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts:126]:
```suggestion
const hasNodeChildren = (children: MarkdownAstNode['children']): children is MarkdownAstNode[] =>
    !!children && Array.isArray(children) && children.length > 0;
```

File: src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts [hunk 4]
```diff
@@ -12,32 +12,23 @@ import {
...
     const linkDestinationKey = options?.linkDestinationKey ?? 'link';
     const imageSourceKey = options?.imageSourceKey ?? 'link';
     const imageCaptionKey = options?.imageCaptionKey ?? 'caption';
 
     let children: Array<DeserializedNode<T>> = [{ text: '' }];
 
-    const nodeChildren = node.children;
-    if (nodeChildren && Array.isArray(nodeChildren) && nodeChildren.length > 0) {
-        children = nodeChildren.flatMap((c: MarkdownAstNode) =>
+    if (hasNodeChildren(node)) {
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts:30]:
```suggestion
    if (hasNodeChildren(node.children)) {
```

File: src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts [hunk 5]
```diff
@@ -12,32 +12,23 @@ import {
...
+        children = node.children!.flatMap((c: MarkdownAstNode) =>
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts:31]:
```suggestion
        children = node.children.flatMap((c: MarkdownAstNode) =>
```
So no `!` required :)

**Review by SamuelAlev — APPROVED** [review]
Left few comments to do before merging

File: src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts (hunk 3 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts:126]: Updated

File: src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts (hunk 4 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts:30]: Updated

File: src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts (hunk 5 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts:31]: Updated

**Pull request merged**